src/main/java/use_cases/add_edit_category_use_case/CategoryUCI.java

```diff
@@ -52,11 +52,12 @@ private double toDouble(Object value) throws NumberFormatException, NullPointerE
      *
      * @param categoryIDAdd CategoryID required for adding a new category to designated monthID MonthlyStorage Object.
      * @return CategoryOD String message indicating success/fail add attempt.
+     * @throws EntityException thrown when the new category input is invalid.
      */
     @Override
-    public CategoryOD addCategoryInMonth(CategoryID categoryIDAdd) {
+    public CategoryOD addCategoryInMonth(CategoryID categoryIDAdd) throws EntityException {
+        this.month = categoryIDAdd.getSession().getMonthlyData(categoryIDAdd.getMonthID());
         try {
-            this.month = categoryIDAdd.getSession().getMonthlyData(categoryIDAdd.getMonthID());
```

Comment:
@luluwuwu what do you think of this change. My reasoning can be seen in the commit message:

 Made addExpenseInMonth and addCategoryInMonth methods throw EntityException to fix the IntelliJ warning that only the interface was throwing but not any actual implementations. Also, I believe this is more correct since the EntityException caused by having the wrong monthID is caught in the view and shouldn't be caught here where the error message says it's because there's a category/expense with the same name.

Reply:
I am not Lulu but this change makes sense to me!

Reply:
I think when Lulu and I talked about this, we said that although such an error should never occur in theory, it may cause the program to terminate if it's not caught anywhere in the code. Thus we decided to put this line of code inside the try catch. However, I do understand that this Exception does not match the warning message, so I'd say we better put it in another try catch that returns the correct warning.

Reply:
(Just moving this comment here)

Right, the monthlyStorage.addExpense() and addCategory() methods throw the EntityException.
But since currently this line is throwing the EntityException as well this.month = session.getMonthlyData(expenseIDAdd.getMonthID()); , should we move this line above the try catch so it doesn't mistake it?

Reply:
Or actually what Yin is saying is more correct

Reply:
Wait, never mind, I see that it gets caught in the view. So I agree is implementation is more correct!

Reply:
> Or actually what Yin is saying is more correct

Lulu, I think we both kind of forgot, but we had a conservation after that, where we said this is a different kind of exception so we want to catch it in the view.

Reply:
I believe before the view is able to catch that Exception, the catch within the use case is going to handle it no?

Reply:
That's why I suggest moving the line with getMonthlyData() outside the try-catch in ExpenseUCI. Let me know if that makes sense.

Reply:
Yep, moving the getMonthlyData outside the try catch is what I did

Reply:
> That's why I suggest moving the line with getMonthlyData() outside the try-catch in ExpenseUCI. Let me know if that makes sense.

I'm not sure I do. I believe it was inside the try catch previously, and now Ari is moving it outside, no? This change will result in the use case interactor throwing the issue directly to the view before it runs any code in the addCategoryInMonth().

Reply:
So does everyone think this change is all right?

Reply:
I don't think that's going to happen? There isn't a try catch for this line so the code does not terminate after throwing the exception. (correct me if I'm wrong) 
I'm rather more concerned that if there is an EntityException thrown, the adding of an Expense is performed but it's an invalid month. However, I think this problem can be safely ignored since we know there shouldn't be an invalid monthID.

Reply:
Yes, I approve of this change.

Reply:
> I don't think that's going to happen? There isn't a try catch for this line so the code does not terminate after throwing the exception. (correct me if I'm wrong) 

As Ari pointe out, if you look at the AddExpenseV and EditExpenseV, there are try catch for catching is kind of exception.

Reply:
Yes, I was thinking it wrong. But I don't think that will be an issue either. 
Since if an EntityException is thrown due to an invalid monthID, we wouldn't want it to enter the use case. And if an EntityException is thrown due to a sameName error, it will be caught by the method within ExpenseUCI before it can be thrown in ExpenseV.

So I will approve this change.